### PR TITLE
Copy che-rest-apis code from che-workspace-operator

### PR DESCRIPTION
**Pull request by amisevsk**
### Description
This PR copies the che-rest-apis subproject from https://github.com/che-incubator/che-workspace-operator to this repo, using the commit https://github.com/che-incubator/che-workspace-operator/commit/ec1d080aa35086e7313cbaccdb3a3daa061a2215 as a base. It consists of three commits:

1. Copy in code without additional changes
2. Add README.md and dockerfile used for building without installing GraalVM (which can be tricky as the version is old and documentation is nowhere)
3. Clean up the codebase slightly.

### CODE REVIEW

**Comment by amisevsk** [comment]
An image built from this PR is available at `amisevsk/che-rest-apis:latest`

**Review by davidfestal — APPROVED** [review]

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Note: this dockerignore is used by both the native binary and final container builds
+target
+!target/*-runner
+!target/libsunec.so
```
Comment (sleshchenko) [inline .dockerignore]:
```suggestion
!target/libsunec.so

```

File: src/main/java/org/eclipse/che/incubator/crd/cherestapis/ApiResource.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    @PUT
+    @Path("workspace/{id}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response update(
+        @PathParam("id") String id,
+        WorkspaceDto update) {
+      return addCorsHeader(apiService.getWorkspace(id));
+    }  
+}
```
Comment (sleshchenko) [inline src/main/java/org/eclipse/che/incubator/crd/cherestapis/ApiResource.java]:
```suggestion
}

```

File: src/test/java/org/eclipse/che/incubator/crd/cherestapis/ApiResourceTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Disabled
+    @Test
+    public void testHelloEndpoint() {
+        given()
+          .when().get("/api/workspace/unkn")
+          .then()
+             .statusCode(404);
+    }
+}
```
Comment (sleshchenko) [inline src/test/java/org/eclipse/che/incubator/crd/cherestapis/ApiResourceTest.java]:
```suggestion
}

```

File: src/test/java/org/eclipse/che/incubator/crd/cherestapis/ApiServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    @Test
+    public void parseDevfile()
+            throws IOException, ServerException, DevfileException, ValidationException, ApiException {
+        InputStream stream = this.getClass().getResourceAsStream("devfiles/petclinic-sample.yaml");
+        String devfileYaml = IOUtils.toString(stream);
+        DevfileImpl devfile = service.parseDevFile(devfileYaml);
+        WorkspaceDto workspace = service.convertToWorkspace(devfile, null);
+        Assertions.assertEquals(workspace.getConfig().getName(), "petclinic");
+    }
+}
```
Comment (sleshchenko) [inline src/test/java/org/eclipse/che/incubator/crd/cherestapis/ApiServiceTest.java]:
```suggestion
}

```

**Review by sleshchenko — APPROVED** [review]

**Pull request merged**